ledger/cryptarchia-ledger/src/lib.rs

```diff
@@ -303,35 +346,49 @@ impl core::fmt::Debug for LedgerState {
 #[cfg(test)]
 pub mod tests {
     use super::{EpochState, Ledger, LedgerState};
-    use crate::{
-        crypto::Blake2b, Commitment, Config, Header, HeaderId, LeaderProof, LedgerError, Nullifier,
-    };
+    use crate::{crypto::Blake2b, Commitment, Config, LeaderProof, LedgerError, Nullifier};
     use blake2::Digest;
     use cryptarchia_engine::Slot;
     use std::hash::{DefaultHasher, Hash, Hasher};
 
-    pub fn header(slot: impl Into<Slot>, parent: HeaderId, coin: Coin) -> Header {
-        let slot = slot.into();
-        Header::new(parent, 0, [0; 32].into(), slot, coin.to_proof(slot))
+    fn update_ledger(
+        ledger: &mut Ledger<[u8; 32]>,
+        parent: [u8; 32],
+        slot: impl Into<Slot>,
+        coin: Coin,
+    ) -> Result<[u8; 32], LedgerError<[u8; 32]>> {
+        update_orphans(ledger, parent, slot, coin, vec![])
+    }
+
+    fn make_id(parent: [u8; 32], slot: impl Into<Slot>, coin: Coin) -> [u8; 32] {
+        Blake2b::new()
+            .chain_update(parent)
+            .chain_update(slot.into().to_be_bytes())
+            .chain_update(coin.sk.to_be_bytes())
+            .chain_update(coin.nonce.to_be_bytes())
+            .finalize()
+            .into()
     }
 
-    pub fn header_with_orphans(
+    fn update_orphans(
+        ledger: &mut Ledger<[u8; 32]>,
+        parent: [u8; 32],
         slot: impl Into<Slot>,
-        parent: HeaderId,
         coin: Coin,
-        orphans: Vec<Header>,
-    ) -> Header {
-        header(slot, parent, coin).with_orphaned_proofs(orphans)
-    }
-
-    pub fn genesis_header() -> Header {
-        Header::new(
-            [0; 32].into(),
-            0,
-            [0; 32].into(),
-            0.into(),
-            LeaderProof::dummy(0.into()),
-        )
+        orphans: Vec<([u8; 32], (u64, Coin))>,
+    ) -> Result<[u8; 32], LedgerError<[u8; 32]>> {
```

Comment:
Nitpick: Maybe we could have a type-alias for `[u8;32]` to specify what exactly is referring to?